exec/cnex/dictionary.c

```diff
@@ -79,12 +79,49 @@ Dictionary *dictionary_createDictionary(int64_t size)
     d->len = 0;
     d->max = size;
     d->data = malloc(d->max * sizeof(struct tagTDictionaryEntry));
-    if (!d->data) {
+    if (d->data == NULL) {
         fatal_error("Could not allocate space for %d dictionary entries.", d->max);
     }
+    //for (int64_t i = 0; i < d->len; i++) {
+    //    DictionaryEntry *e = &d->data[i];
+    //    cell_clearCell(e->value);
+    //    string_clearString(e->key);
+    //}
+
+    return d;
+}
+
+Dictionary *dictionary_copyDictionary(Dictionary *self)
+{
+    Dictionary *d = dictionary_createDictionary(self->len);
+    //d->len = self->len;
```

Comment:
What's this line for? Seems wrong.

Reply:
Which line?  The commented out one, or the create_Dictionary() line?

Reply:
The commented one. Why is that there? It would make no sense if it were uncommented, and as a comment it doesn't really help.

Reply:
OH.  Right, old code.  Shouldn't be there.